src/main/java/com/onarandombox/MultiverseCore/commands/DumpsCommand.java

```diff
@@ -0,0 +1,320 @@
+        }
+
+
+        // Deal with default case of logs
+        LogsType logsType = parsedFlags.flagValue("--logs", LogsType.class);
+        if (logsType == null) {
+            logsType = LogsType.MCLOGS;
+            Logging.finer("Logs was null so set to mclogs");
+        }
+
+
+
+
+        // Deal with default case of upload
+        Services services = parsedFlags.flagValue("--upload", Services.class);
+        if (services == null) {
+            services = Services.PASTEGG;
+            Logging.finer("Upload was null so set to pastegg");
+        }
+
+
+        // Need to be final for some reason...
+        final LogsType finalLogsType = logsType;
+        final Services finalServices = services;
+        BukkitRunnable logPoster = new BukkitRunnable() {
+            @Override
+            public void run() {
+                HashMap<String, String> pasteURLs = new HashMap<>();
+                Logging.finer("Logs type is: " + finalLogsType);
+                Logging.finer("Services is: " + finalServices);
+
+                // Deal with logs flag
+                if (!parsedFlags.hasFlag("--paranoid")) {
+                    switch (finalLogsType) {
+                        case MCLOGS -> {
+                            issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING_LOGS, "{link}", "https://mclo.gs");
+                            pasteURLs.put("Logs", postToService(PasteServiceType.MCLOGS, false, getLogs(), null));
+                        }
+                        case APPEND -> {
+                            versionEvent.putDetailedVersionInfo("latest.log", getLogs());
+                        }
+                    }
+                }
+
+                // Get the files from the event
+                final Map<String, String> files = versionEvent.getDetailedVersionInfo();
+
+                // Deal with uploading debug info
+                switch (finalServices) {
+                    case PASTEGG -> {
+                        issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING, "{link}", "https://paste.gg");
+                        pasteURLs.put("paste.gg", postToService(PasteServiceType.PASTEGG, true, null, files));
+                    }
+                    case PASTESDEV -> {
+                        issuer.sendInfo(MVCorei18n.DUMPS_UPLOADING, "{link}", "https://pastes.dev");
+                        pasteURLs.put("pastes.dev", postToService(PasteServiceType.PASTESDEV, true, null, files));
+                    }
+                }
+
+                // Finally, loop through and print all URLs
+                for (String service : pasteURLs.keySet()) {
+                    String link = pasteURLs.get(service);
+                    issuer.sendInfo(MVCorei18n.DUMPS_URL_LIST, "{service}", service, "{link}", link);
+                }
+
+            }
+        };
+
+        // Run the uploader async as it could take some time to upload the debug info
+        logPoster.runTaskAsynchronously(plugin);
+    }
```

Comment:
This method is nearly 90 lines long (way too long). Could we extract some of this, like especially the runnable stuff, into a separate method/class?

Reply:
Regarding moving `logPoster` to its own class, where would that be placed as it is only really used in `DumpsCommand.java` so `utils/` wouldn't make sense imo.

Reply:
I think separating to methods is fine. Looks to me like you can split "Deal with logs flag" and "Deal with uploading debug info" into separate methods.